knip.ts

```diff
@@ -0,0 +1,136 @@
+import type { KnipConfig } from 'knip';
+
+export default {
+  rules: {
+    classMembers: 'off',
+    duplicates: 'off',
+    enumMembers: 'off',
+    exports: 'off',
+    nsExports: 'off',
+    nsTypes: 'off',
+    types: 'off',
+    unresolved: 'off',
+  },
+  workspaces: {
+    '.': {
+      ignoreDependencies: [
+        '@babel/code-frame',
+        '@babel/core',
+        '@babel/eslint-parser',
+        '@babel/parser',
+        '@babel/types',
+        '@nx/workspace',
+        'cross-fetch',
+        'glob',
+        'husky',
+        'jest-specific-snapshot',
+        'make-dir',
+        'ncp',
+        'tmp',
+
+        // imported in eslint.config.js
+        '@typescript-eslint/utils',
+        // imported in eslint.config.mjs
+        '@typescript-eslint/eslint-plugin-internal',
+      ],
+      entry: ['tools/release/changelog-renderer.js'],
+      ignoreBinaries: [
+        // https://github.com/webpro/knip/issues/433
+        'stylelint',
+      ],
+    },
+    'packages/ast-spec': {
+      ignore: [
+        'src/**/fixtures/**',
+        'tests/*.type-test.ts',
+        // @typescript-eslint/typescript-estree is not listed in dependencies to avoid circular dependency errors
+        // You can check a more detailed explanation in this file
+        'tests/util/parsers/typescript-estree-import.ts',
+      ],
+    },
+    'packages/eslint-plugin': {
+      ignore: ['tests/fixtures/**'],
+    },
+    'packages/eslint-plugin-internal': {
+      ignore: ['tests/fixtures/**'],
+    },
+    'packages/integration-tests': {
+      ignore: ['fixtures/**'],
+    },
+    'packages/parser': {
+      ignore: ['tests/fixtures/**'],
+    },
+    'packages/repo-tools': {
+      ignoreDependencies: ['tsconfig.json'],
+    },
+    'packages/scope-manager': {
+      ignore: ['tests/fixtures/**'],
+    },
+    'packages/type-utils': {
+      ignore: ['tests/fixtures/**'],
+    },
+    'packages/typescript-estree': {
+      entry: ['src/use-at-your-own-risk.ts'],
+      ignore: ['tests/fixtures/**'],
+    },
+    'packages/website': {
+      entry: [
+        'docusaurus.config.mts',
+        'src/pages/**/*.tsx',
+
+        // imported in MDX docs
+        'src/components/RulesTable/index.tsx',
+        'src/components/TypeScriptOverlap/index.tsx',
+        'src/components/team/TeamBioList.tsx',
+        'src/components/team/TeamBioDivider.tsx',
+
+        // used by Docusaurus
+        'src/theme/CodeBlock/Content/String.tsx',
+        'src/theme/MDXComponents/index.tsx',
+        'src/theme/NotFound/Content/index.tsx',
```

Comment:
Maybe, so we don't have to keep a manual list? (pending Knip Docusaurus support)

```suggestion
        // imported in MDX docs
        'src/components/**/*.tsx',

        // used by Docusaurus
        'src/theme/**/index.tsx',
```